sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpConnection.java

```diff
@@ -57,6 +58,17 @@
      */
     Mono<AmqpSession> createSession(String sessionName);
 
+    /**
+     * Creates a new session with the given session name. It will set up {@link Coordinator} as needed for transactions.
+     *
+     * @param sessionName Name of the session to be created.
+     * @param coordinatorRequired If coordinator is required for this session.
+     * @return The AMQP session that was created.
+     *
+     * @see <a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transactions-v1.0-os.html#section-coordination">Transactions-Coordinator</a>
```

Comment:
Transaction Coordinator instead of the hyphen

Reply:
This is a very specific API that is only used to support transactions. As an end user, I don't care about how transactions are coordinated or that there is even a coordinator.

```
Mono<AmqpSession> createTransaction(String name);
```

Reply:
We are not exposing this API to SB user.  But as far as core-amqp library is concerned,  this concept `Coordinator` is part of AMQP  spec. So the api `createSession(sessoinName, boolean coordinatorRequired)` is added. 

The `AMQP Session`  is created created in `core-amqp` library and `AMQP Coordinator` needs to be setup as first link under this Session. So when we create `AMQP Session`, we setup this session with `Coordinator link`.